native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/dsl/NativeImageOptions.java

```diff
@@ -91,6 +92,7 @@
      * @return mainClass The main class.
      */
     @Input
+    @Optional
```

Comment:
:x: This change alone isn't sufficient. The code needs to be updated to remove the default value, and test if a value was set instead.

Reply:
Default value for `mainClass` property is set only when related property in the `application` block is set, as seen here:
https://github.com/graalvm/native-build-tools/blob/1456c696a3782ff2dfaf4c76fb225095aa3b8b0a/native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/NativeImagePlugin.java#L132-L134

In other cases this value would be omitted, as expected.
https://github.com/graalvm/native-build-tools/blob/1456c696a3782ff2dfaf4c76fb225095aa3b8b0a/native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/internal/NativeImageCommandLineProvider.java#L134-L136

IMO this will handle all of the use cases.

Reply:
:+1: